Assets/SEE/Game/SceneManipulation/SpatialMetrics.cs

```diff
@@ -0,0 +1,253 @@
+﻿using System;
+using UnityEngine;
+
+namespace SEE.Game.SceneManipulation
+{
+    /// <summary>
+    /// Stores metrics that are used for working with node placement.
+    /// </summary>
+    public class SpatialMetrics
+    {
+        /// <summary>
+        /// A small offset used for placement just outside specific bounds to prevent Z-fighting.
+        /// </summary>
+        public const float PlacementOffset = 0.0001f;
+
+        /// <summary>
+        /// The default local X/Z scale of a (new) node placed on another node.
+        /// </summary>
+        public static float DefaultNodeLocalScale = 0.2f;
+
+        /// <summary>
+        /// Half the default local X/Z scale of a (new) node placed on another node,
+        /// typically used in bounds.
+        /// </summary>
+        public static float HalfDefaultNodeLocalScale = DefaultNodeLocalScale / 2f;
+
+        /// <summary>
+        /// The default world-space Y-axis size of a (new) node.
+        /// </summary>
+        public static float DefaultNodeHeight = 0.002f;
+
+        /// <summary>
+        /// The minimal size of a node in world space.
+        /// </summary>
+        public static readonly Vector3 MinNodeSize = new (0.06f, 0.001f, 0.06f);
+
+        /// <summary>
+        /// The minimal world-space distance between nodes for placing or resizing (x/z, y).
+        /// </summary>
+        public static readonly Vector3 Padding = new(0.004f, 0.0001f, 0.004f);
+
+        /// <summary>
+        /// The minimal size of a node in local scale.
+        /// </summary>
+        /// <param name="localTransform">Transform of the reference node.</param>
+        /// <returns>The <see cref="MinNodeSize"/> in <paramref name="localTransform"/>'s local scale.</returns>
+        public static Vector3 MinNodeSizeLocalScale(Transform localTransform)
+        {
+            return localTransform.InverseTransformVector(MinNodeSize);
+        }
+    }
+
+    /// <summary>
+    /// Direction in 2D plane.
+    /// </summary>
+    public enum Direction2D
+    {
+        Left, Right, Back, Front, None
+    }
+
+    /// <summary>
+    /// Data structure for 2-dimensional bounds.
+    /// </summary>
+    public struct Bounds2D
```

Comment:
Perhaps a `record` would be a better fit here. In that case, we'd only need to write:
```csharp
    public record Bounds2D(float Left, float Right, float Back, float Front)
```
And would automatically get those fields, a constructor taking them as parameters, a ToString method, and an Equals implementation.

Reply:
A record struct has init-only attributes. The current implementation allows the fields to be changed after init.